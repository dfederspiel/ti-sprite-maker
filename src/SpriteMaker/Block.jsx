import React from "react";

const Block = (props) => {

  return (
    <div
      style={{
        backgroundColor: props.state ? `#${props.color}` : "white",
        width: "30px",
        height: "30px",
        margin: "2px",
        transition: "all .5s",
      }}
      className={props.state ? "on" : "off"}
      onClick={() => {
        props.clicked()
      }}
      onMouseOver={() => {
          props.clicked()
      }}
    ></div>
  );
};

export default Block;
