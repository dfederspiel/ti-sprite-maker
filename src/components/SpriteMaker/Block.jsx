import React from 'react';
import PropTypes from 'prop-types';

const Block = ({ color, state, clicked }) => (
  // eslint-disable-next-line jsx-a11y/control-has-associated-label
  <button
    type="button"
    data-testid="block"
    style={{
      border: 'none',
      backgroundColor: state ? `#${color}` : 'white',
      width: '30px',
      height: '30px',
      margin: '2px',
      transition: 'all .5s',
    }}
    className={state ? 'on' : 'off'}
    onClick={() => {
      clicked();
    }}
  />
);

Block.propTypes = {
  /**
   * indicates the toggle state of the block
   */
  state: PropTypes.bool,
  /**
   * hexadecimal RGB value
   */
  color: PropTypes.string,
  /**
   * callback handler for click events
   */
  clicked: PropTypes.func,
};

Block.defaultProps = {
  state: false,
  color: '000',
  clicked: null,
};

export default Block;
