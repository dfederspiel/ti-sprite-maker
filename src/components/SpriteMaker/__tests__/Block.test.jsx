import React from 'react';
import { act, fireEvent, render } from '@testing-library/react';
import Block from '../Block';

describe('the block', () => {
  it('defaults to off', () => {
    const { asFragment } = render(<Block />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('can initialize to on', () => {
    const { asFragment } = render(<Block state />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('can initialize with a color', () => {
    const { asFragment } = render(<Block state color="369" />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('when the block is clicked', () => {
    it('fires the click handler', async () => {
      let timesClicked = 0;
      const { getByTestId } = render(<Block clicked={() => {
        // eslint-disable-next-line no-plusplus
        timesClicked++;
      }}
      />);
      const block = getByTestId('block');
      act(() => {
        fireEvent.click(block);
        expect(timesClicked).toEqual(1);
        fireEvent.click(block);
        expect(timesClicked).toEqual(2);
      });
    });
  });
});
