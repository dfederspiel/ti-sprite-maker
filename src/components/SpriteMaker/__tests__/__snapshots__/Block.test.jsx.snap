// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the block can initialize to on 1`] = `
<DocumentFragment>
  <button
    class="on"
    data-testid="block"
    style="background-color: rgb(0, 0, 0); width: 30px; height: 30px; margin: 2px; transition: all .5s;"
    type="button"
  />
</DocumentFragment>
`;

exports[`the block can initialize with a color 1`] = `
<DocumentFragment>
  <button
    class="on"
    data-testid="block"
    style="background-color: rgb(51, 102, 153); width: 30px; height: 30px; margin: 2px; transition: all .5s;"
    type="button"
  />
</DocumentFragment>
`;

exports[`the block defaults to off 1`] = `
<DocumentFragment>
  <button
    class="off"
    data-testid="block"
    style="background-color: white; width: 30px; height: 30px; margin: 2px; transition: all .5s;"
    type="button"
  />
</DocumentFragment>
`;
